custom-sign-in/app/src/main/java/com/okta/android/samples/custom_sign_in/rx_client/RxSessionClientImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License,
+ * Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the
+ * License.
+ */
+
+package com.okta.android.samples.custom_sign_in.rx_client;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.okta.oidc.OIDCConfig;
+import com.okta.oidc.OktaState;
+import com.okta.oidc.Tokens;
+import com.okta.oidc.clients.sessions.SyncSessionClient;
+import com.okta.oidc.clients.sessions.SyncSessionClientFactory;
+import com.okta.oidc.net.HttpConnection;
+import com.okta.oidc.net.HttpConnectionFactory;
+import com.okta.oidc.net.response.IntrospectInfo;
+import com.okta.oidc.net.response.UserInfo;
+
+import org.json.JSONObject;
+
+import java.util.Map;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+
+class RxSessionClientImpl implements RxSessionClient {
+    private SyncSessionClient mSyncSessionClientImpl;
+
+    RxSessionClientImpl(OIDCConfig oidcConfig, OktaState oktaState,
+                        HttpConnectionFactory connectionFactory) {
+        mSyncSessionClientImpl = new SyncSessionClientFactory().createClient(oidcConfig, oktaState, connectionFactory);
```

Comment:
It looks like `RxSessionClientImpl` just wraps around `SyncSessionClient`. Can't you pass in the `SyncSessionClient` from `mSyncAuthClient`? Since you already created `mSyncAuthClient` and it contains a `SyncSessionClient`.

Also why do we use `SyncSessionClientFactory` to create `SyncSessionClientImpl` and not `SessionClientImpl`? It is a bit inconsistent. `SyncSessionClientFactory implements ClientFactory` which is used in `OktaBuilder`. This allows developers to use the OktaBuilder and add
`withAuthenticationClientFactory(SyncSessionClientFactory)` which is wrong since they will get a session client instead of a Auth client. So can we remove `SyncSessionClientFactory` in the oidc library?

Reply:
I think yes. 
I created factories to make workable previous version of interface, where you can manage which clients create by using related factory. Now, the reason of factory lost, or we can make them package-private and use existing builders in Okta class to make ability for clients create their own implementation of clients